backend/src/test/java/com/redhat/cloud/notifications/auth/rbac/workspace/RbacWorkspacesOidcClientTest.java

```diff
@@ -0,0 +1,39 @@
+package com.redhat.cloud.notifications.auth.rbac.workspace;
+
+import com.redhat.cloud.notifications.auth.rbac.RbacWorkspacesOidcClient;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Integration test for RbacWorkspacesOidcClient that verifies OIDC authentication headers
+ * are properly added to HTTP requests. The RBAC server mock returns 401 for missing/invalid
+ * Authorization headers and 200 for correct ones.
+ */
+@QuarkusTest
+@QuarkusTestResource(OidcServerMockResource.class)
+@QuarkusTestResource(RbacServerMockResource.class)
+public class RbacWorkspacesOidcClientTest {
+
+    @Inject
+    @RestClient
+    RbacWorkspacesOidcClient rbacClient;
+
+    private static final String TEST_ORG_ID = "test-org-id";
+
+    @Test
+    @DisplayName("Should successfully call getWorkspaces with OIDC authentication")
+    void shouldSuccessfullyCallGetWorkspaces() {
+        // If OIDC headers are missing, the mock server will return 401 and this will throw an exception
+        // If OIDC headers are present and correct, the mock server returns 200 and this succeeds
```

Comment:
Do we have some tests somewhere for incorrect OIDC header, OIDC issuer timeout or OIDC issuer errors like 500, 401, 403?

Reply:
My concern here is more to check how Notifications will handle different kind of errors/exceptions that Quarkus OIDC extention could return, will notifiations will return a 404 or 403 if unable get a proper RBAC response because of an OIDC issue?

Reply:
I created https://issues.redhat.com/browse/RHCLOUD-41298 to further investigate this.